src/GenericContainer/include/BipedalLocomotionControllers/GenericContainer.h

```diff
@@ -0,0 +1,289 @@
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_GENERICCONTAINER_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_GENERICCONTAINER_H
+
+#include <iDynTree/Core/Span.h>
+
+#include <BipedalLocomotionControllers/TemplateHelpers.h>
+
+#include <functional>
+#include <cassert>
+#include <iostream>
+
+namespace BipedalLocomotionControllers {
+    template <typename T>
+    class GenericContainer;
+}
+
+template <typename T>
+class BipedalLocomotionControllers::GenericContainer
+{
+public:
+
+    using element_type = typename iDynTree::Span<T>::element_type;
+    using value_type = typename iDynTree::Span<T>::value_type;
+    using index_type = typename iDynTree::Span<T>::index_type;
+    using pointer = typename iDynTree::Span<T>::pointer;
+    using reference = typename iDynTree::Span<T>::reference;
+
+    using iterator = typename iDynTree::Span<T>::iterator;
+    using const_iterator = typename iDynTree::Span<T>::const_iterator;
+    using reverse_iterator = typename iDynTree::Span<T>::reverse_iterator;
+    using const_reverse_iterator = typename iDynTree::Span<T>::const_reverse_iterator;
+
+    using size_type = typename iDynTree::Span<T>::size_type;
+    using resize_function_type = std::function<iDynTree::Span<T>(index_type)>;
+
+private:
+    iDynTree::Span<T> m_span;
+    resize_function_type m_resizeLambda;
+
+public:
+
+    GenericContainer() = delete; //Since the = copies the content, there would be no possibility to set the internal pointers.
+
+    GenericContainer(iDynTree::Span<T> span, resize_function_type resizeLambda)
+    {
+        m_span = span;
+        m_resizeLambda = resizeLambda;
+    }
+
+    GenericContainer(iDynTree::Span<T> span)
+    {
+        m_span = span;
+        m_resizeLambda = [span](index_type size){return span;};
+    }
+
+    ~GenericContainer() = default;
+
+    GenericContainer(const GenericContainer<T>& other) = delete; //It would not be clear if you are copying the pointers or the content.
+
+    GenericContainer(GenericContainer<T>&& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    bool clone(const GenericContainer<T>& other)
+    {
+        if (size() != other.size())
+        {
+            if (!resize(other.size()))
+            {
+                std::cerr << "[Generic container] Failed to resize. Copy aborted" << std::endl;
+                return false;
+            }
+        }
+
+        for (index_type i = 0; i < size(); ++i)
+        {
+            this->operator[](i) = other[i];
+        }
+
+        return true;
+    }
+
+    void operator=(const GenericContainer<T>& other)
+    {
+        clone(other);
+    }
+
+    void operator=(GenericContainer<T>&& other)
+    {
+        clone(other);
+    }
+
+
+    bool resize(index_type newSize)
+    {
+        m_span = m_resizeLambda(newSize);
+        return m_span.size() == newSize;
+    }
```

Comment:
Just a small comment In general, `resize()` doesn't return a `boolean`.
I completely understand the reason why you add the `bool`. I wonder if we should also change the logic in `YarpUtilities`
https://github.com/dic-iit/bipedal-locomotion-controllers/blob/e055599aa3cbe6bc3f4976a53c358694a5c0970a/src/YarpUtilities/include/BipedalLocomotionControllers/YarpUtilities/Helper.tpp#L134-L135

Probably `YarpUtilities` may support only `GenericContainer`

Reply:
This is something I did not think of. I may use a different name for ``resize`` (like ``resizeContainer``), or I may add another case where I check first if the input is a ``GenericContainer`` and act accordingly, or both. What would you prefer?

Reply:
Probably the easiest way is to change the `YarpUtilities` component
Furthermore, `getVectorFromSearchable` wants as input a reference to `T`
```cpp
template <typename T>
bool getVectorFromSearchable(const yarp::os::Searchable& config, const std::string& key, T& vector);
```
Since we are moving towards the use of  `GenericContainer` probably we can change `YarpUtilities` as
```cpp
template <typename T>
bool getVectorFromSearchable(const yarp::os::Searchable& config, const std::string& key, GenericContainer<T> vector);

bool getVectorFromSearchable(const yarp::os::Searchable& config, const std::string& key, std::vector<bool>& vector);
```

Adding a **HUGE** disclaimer in the documentation saying that `GenericContainer<T>`  is passed by copy because it contains a `span` object.

Reply:
Notice that I explicitly deleted the copy operator. I wanted to avoid ambiguity on the copy: is the span copied or the content?
I then decided that the span is copied only in the move constructor, while the assignment operator copies the content into the pointed container. In order to copy the content, I need to make sure that I am pointing to existing containers. Clearly, in the copy constructor, this is not possible. 
We are left to decide if we want the copy constructor to copy the ``span``. I would prefer to leave it deleted to avoid confusion. What do you think?

Reply:
I dealt with https://github.com/dic-iit/bipedal-locomotion-controllers/pull/29#discussion_r395166265 in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/29/commits/6bea0ccfe36aa1aa1005ebf893b2b4815d6f956e